docs/rfc/rfc-020-block-structure-consolidation.md

```diff
@@ -0,0 +1,364 @@
+  bytes                        evidence_hash        = 13;
+  bytes                        proposer_address     = 14;
+}
+
+```
+
+```proto
+message Data {
+  repeated bytes txs = 1;
+}
+```
+
+```proto
+message EvidenceList {
+  repeated Evidence evidence = 1;
+}
+```
+
+```proto
+message Commit {
+  int64              height     = 1;
+  int32              round      = 2;
+  BlockID            block_id   = 3;
+  repeated CommitSig signatures = 4;
+}
+```
+
+```proto
+message CommitSig {
+  BlockIDFlag               block_id_flag     = 1;
+  bytes                     validator_address = 2;
+  google.protobuf.Timestamp timestamp         = 3;
+  bytes                     signature         = 4;
+}
+```
+
+```proto
+message BlockID {
+  bytes         hash            = 1;
+  PartSetHeader part_set_header = 2;
+}
+```
+
+### On Tendermint Blocks
+
+#### What is a Tendermint 'Block'?
+
+A block is the structure produced as the result of an instance of the Tendermint
+consensus algorithm. At its simplest, the 'block' can be represented as a Merkle
+root hash of all of the data used to construct and produce the hash. Our current
+block proto structure includes _far_ from all of the data used to produce the
+hashes included in the block.
+
+It does not contain the full `AppState`, it does not contain the `ConsensusParams`,
+nor the `LastResults`, nor the `ValidatorSet`. Additionally, the layout of
+the block structure is not inherently tied to this Merkle root hash. Different
+layouts of the same set of data could trivially be used to construct the
+exact same hash. The thing we currently call the 'Block' is really just a view
+into a subset of the data used to construct the root hash. Sections of the
+structure can be modified as long as alternative methods exist to query and
+retrieve the constituent values.
+
+#### Why this digression?
+
+This digression is aimed at informing what it means to consolidate 'fields' in the
+'block'. The discussion of what should be included in the block can be teased
+apart into a few different lines of inquiry.
+
+1. What values need to be included as part of the Merkle tree so that the
+consensus algorithm can use proof-of-stake consensus to validate all of the
+properties of the chain that we would like?
+2. How can we create views of the data that can be easily retrieved, stored, and
+verified by the relevant protocols?
+
+These two concerns are intertwined at the moment as a result of how we store
+and propagate our data but they don't necessarily need to be. This document
+focuses primarily on the first concern by suggesting fields that can be
+completely removed without any loss in the function of our consensus algorithm.
+
+This document also suggests ways that we may update our storage and propagation
+mechanisms to better take advantage of Merkle tree nature of our data although
+these are not its primary concern.
+
+## Discussion
+
+### Data to consider removing
```

Comment:
I went through the data structures myself and I think that the following fields in `Commit`

* height
* round
* block_id

are redundant, and could therefore be removed.

They are redundant in that they can be "gleaned" from the rest of the fields in the block.